backend/service/k8s/clientset.go

```diff
@@ -142,6 +143,14 @@ func (m *managerImpl) Clientsets(ctx context.Context) (map[string]ContextClients
 	return ret, nil
 }
 
+func (m *managerImpl) GetK8sClientsetSimple(clientset string) (ContextClientset, error) {
```

Comment:
It seems like this should be `GetK8sClientset` and the function below should be `GetK8sClientsetWithFallback` but I know it would be a lot of updates to make. Maybe we can add some docstrings to denote the differences to consumers?

Reply:
Yeah I'm not in favor of breaking that API at the moment, but ill add some comments to clarify its usage.